Update best practices with new section. (#3124)
couchbase/docs-server: modules/eventing/pages/troubleshooting-best-practices.adoc

----
mystring.match(/(\S+)[^=]=[\"\']?((?:.(?![\"\']?\s+(?:\S+)[^=]=|[>\"\']))+.)[\"\']?/g);
----

> == Why do some of my functions dealing with large datasets take so long to undeploy?

Reviewer: ```suggestion
== Why does my function take so long to undeploy?
```

> Undeployment depends on the size of the Eventing metadata collection (or scratchpad) and not the source or destination. 

Reviewer: ```suggestion
Undeployment time depends on the size of the Eventing metadata collection or scratchpad, not the source or destination. 
```

> If any Eventing Function is undeployed on a collection with lots of tombstones the Eventing service has to re-stream these documents for any Eventing Function during undeployment phase. 

Reviewer: ```suggestion
If you undeploy an Eventing Function on a collection with lots of tombstones, the Eventing service has to re-stream these documents during the undeployment phase. 
```

> If we have an Eventing function that generates a billion timers which have all fired (relatively quickly say an hour or two) the Eventing Function might take minutes or even an hour to undeploy due to the tombstones of deleted items in this case timers remaining on disk. 

Reviewer: ```suggestion
If you have an Eventing function that generates a billion timers, which have all fired quickly, the Eventing Function might take minutes or an hour to undeploy because of the tombstones remaining on disk. 
```

> In fact any Eventing Function that shares the same metadata collection will also experience the same slow undeployment.

Reviewer: ```suggestion
Any Eventing Function that shares the same metadata collection will also be slow to undeploy.
```

> Workarounds (best practice) is to have a separate metadata collection for non-timer Eventing Functions which will stream 1024 documents each. 

Reviewer: ```suggestion
It's recommended to have a separate metadata collection for non-timer Eventing Functions, which will stream 1024 documents each. 
```

> Eventing Functions that generate a massive amount of timers (1000+/sec.) should use a private collection for metadata and not share this area with other Eventing Functions.  

Reviewer: ```suggestion
An Eventing Function that generates more than 1000 timers per second should use a private collection for metadata and not share this area with other Eventing Functions.  
```

> Note the tombstones will remain by default for 3 days unless the Metadata Purge Interval is adjusted. 

Reviewer: ```suggestion
By default, tombstones remain on disk for 3 days. 
Adjust the Metadata Purge Interval to reduce the number of accumulated tombstones. 
```

> By lowering the bucket's Metadata Purge Interval  (it can be set to a low as 0.4 hours) the potential for accumulating a massive number of tombstones can be avoided.

Reviewer: ```suggestion

```
